.github/workflows/run_tests.yml

```
@@ -0,0 +1,32 @@
+name: Run Unit Test via Pytest
+
+on:
+  # trigger on pushes to any branch, but not main
+  push:
+    branches-ignore:
+      - main
+  # and also on PRs to main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+      - name: Test with pytest
+        run: |
+          pytest  -v -s
```

Should we cache the downloaded dataset to avoid re-downloads on every push? Or is that not necessary?
```suggestion
          pytest  -v -s
      - name: Cache data
        uses: actions/cache@v2
        with:
        path: data
        key: ${{ runner.os }}-data-${{ hashFiles('requirements.txt') }}
        restore-keys: |
          ${{ runner.os }}-data-
```

---

good idea, although the caching should maybe be done on some identifier unique to the dataset. If you're with the testdata you have on AWS now @SimonKamuk you could maybe change the cache key to `meps-reduced-example-data-v0.1.0` or something like that, rather than based the key on the contents of `requirements.txt`